### PR TITLE
Fix: Form submission: zero is not empty

### PR DESCRIPTION
**Pull request by kuzmany**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | 
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/5113
| BC breaks? | 
| Deprecations? | 

This PR was donated with love by Webmecanik.

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Form submission not send form with 0 in value.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:

Note:  On actual staging or 2.11 beta need also this bugfix https://github.com/mautic/mautic/pull/5178 for properly test

1. Create a form
2. Create a bool field or integer field field, and map it to your custom field.
3. Take the values already created in your custom field
4. Select required = YES for this field
5. Fill the form and select values with 0 
6. Submit the form
7. Form not send: fields are required

#### Steps to test this PR:
1. Repeat all steps
2. Validation passed because 0 is also form value

### CODE REVIEW

**Comment by npracht** [comment]
Tested on 2.10.1 and it works !!

File: app/bundles/FormBundle/Model/SubmissionModel.php [hunk 1]
```diff
@@ -249,7 +249,7 @@ public function saveSubmission($post, $server, Form $form, Request $request = nu
                 }
             }
 
-            if (empty($value) && $f->isRequired()) {
+            if ($value == '' && $f->isRequired()) {
```
Comment (alanhartless) [inline app/bundles/FormBundle/Model/SubmissionModel.php]: Should this be a triple equals?

Comment (kuzmany) [reply app/bundles/FormBundle/Model/SubmissionModel.php]: Fixed

**Comment by javjim-mautic** [comment]
works properly

**Comment by npracht** [comment]
Just tested it works !

**Comment by Maxell92** [comment]
@alanhartless Could you approve this PR?

**Pull request merged**